py/server/deephaven/table_listener.py

```diff
@@ -393,3 +398,159 @@ def listen(t: Table, listener: Union[Callable, TableListener], description: str
                                                                     description=description)
     table_listener_handle.start(do_replay=do_replay, replay_lock=replay_lock)
     return table_listener_handle
+
+
+class _ListenerRecorder (JObjectWrapper):
+    """A ListenerRecorder object records the table updates and notifies the associated MergedListener that a change
+    has occurred."""
+
+    j_object_type = _JListenerRecorder
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_listener_recorder
+
+    def __init__(self, table: Table):
+        if not table.is_refreshing:
+            raise DHError(message="table must be a refreshing table.")
+
+        self.j_listener_recorder = _JListenerRecorder("Python Wrapped Listener recorder", table.j_table, None)
+        self.table = table
+
+    def table_update(self) -> Optional[TableUpdate]:
+        """Gets the table update from the listener recorder. If there is no update in the current update graph cycle,
+        returns None.
+
+        Returns:
+            a TableUpdate or None
+        """
+        j_table_update = self.j_listener_recorder.getUpdate()
+        return TableUpdate(self.table, j_table_update) if j_table_update else None
+
+
+class MergedListener(ABC):
+    """A MergedListener has a collection of ListenerRecorder. Each one must complete before the merged listener
+    executes only once for the current update graph cycle. """
```

Comment:
Not a good pydoc string.  It is referring to a private class.

```suggestion
    """An abstract multi-table listener class that should be subclassed by any user multi-table listener class."""
```